QuizOverflow.Data/GenericRepository.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using QuizOverflow.Data.Contracts;
+using QuizOverflow.Models.Abstracts;
+using System.Linq.Expressions;
+
+namespace QuizOverflow.Data
+{
+    public class GenericRepository<T>: IGenericRepository<T> where T : EntityBase
+    {
+        private readonly IQuizOverflowDbContext _context;
+        private DbSet<T> _dbSet;
+
+        public GenericRepository(IQuizOverflowDbContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(
+                    "An instance of QuizOverflowDbContext is required to use this repository.");
+            }
+
+            _context = context;
+            _dbSet = _context.Set<T>();
+        }
+
+        /// <summary>
+        /// Creates a query for taking elements which have not been flagged as deleted.
+        /// </summary>
+        public IQueryable<T> Get()
+        {
+            return _dbSet.Where(e => !e.IsDeleted).AsQueryable();
+        }
+
+        /// <summary>
+        /// Creates a query with a given expression for additional filtering or for getting related
+        /// entities through the navigational properties.
+        /// </summary>
+        public IQueryable<T> GetWhere(Expression<Func<T, bool>> expression)
+        {
+            return _dbSet.Where(expression);
+        }
+
+        /// <summary>
+        /// Prepares the entity to be created with initial CreatedOn, ModifiedOn, IsDeleted properties.
+        /// </summary>
+        public void Create(T entity)
+        {
+            CheckIfEntityIsNull(entity);
+
+            EntityEntry entry = _context.Entry(entity);
+
+            entity.ModifiedOn = DateTime.Now;
+            entity.IsDeleted = false;
+
+            if (entry.State != EntityState.Detached)
+            {
+                entry.State = EntityState.Added;
+            }
+            else
+            {
+                entity.CreatedOn = DateTime.Now;
+            }
```

Comment:
Please add the new entity to the DB set, so it can be tracked by EF.

Reply:
Refactored